bin/pre-install.js

```diff
@@ -0,0 +1,116 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ *  * License, v. 2.0. If a copy of the MPL was not distributed with this
+ *   * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*
+ * This file is run when a user runs `yarn install`, before doing anything else.
+ * We check that various tools we need have the correct version. For now we
+ * check Node and Yarn. */
+
+// The "userAgent" is something like 'yarn/0.28.1 npm/? node/v8.4.0 linux x64'
+// when ran with yarn. With npm it's similar, without the `yarn` part of course.
+const userAgent = process.env.npm_config_user_agent;
+if (!userAgent) {
+  console.error('Error: this script cannot be run directly.');
+  process.exit(-1);
+}
+
+const agents = userAgent.split(' ').reduce((agents, agent) => {
+  const [key, value] = agent.split('/');
```

Comment:
I realize that `const` and destructuring assignments do not work in older nodes. What do you think ?